procfs-core/src/crypto.rs

```diff
@@ -0,0 +1,570 @@
+        Self { fields }
+    }
+}
+
+fn parse_line<T: Iterator<Item = Result<String, std::io::Error>>>(
+    iter: &mut T,
+    to_find: &str,
+    name: &str,
+) -> ProcResult<String> {
+    let line = expect!(iter.next())?;
+    let (key, val) = expect!(line.split_once(':'));
+    if key.trim() != to_find {
+        return Err(build_internal_error!(format!(
+            "could not locate {to_find} in /proc/crypto, block {name}"
+        )));
+    }
+    Ok(val.trim().to_string())
+}
+
+fn parse_fips<T: Iterator<Item = Result<String, std::io::Error>>>(
+    iter: &mut Peekable<T>,
+    name: &str,
+) -> ProcResult<bool> {
+    if iter
+        .peek()
+        .map(|line| line.as_ref().is_ok_and(|line| line.contains("fips")))
+        .unwrap_or(false)
+    {
+        let fips = parse_line(iter, "fips", name)?;
+        if fips == "yes" {
+            return Ok(true);
+        }
+    }
+    Ok(false)
+}
+
+fn parse_bool<T: Iterator<Item = Result<String, std::io::Error>>>(
+    iter: &mut T,
+    to_find: &str,
+    name: &str,
+) -> ProcResult<bool> {
+    match parse_line(iter, to_find, name)?.as_str() {
+        "yes" => Ok(true),
+        "no" => Ok(false),
+        _ => Err(build_internal_error!(format!(
+            "{to_find} for {name} was unrecognised term"
+        ))),
+    }
+}
+
+fn parse_gen_iv<T: Iterator<Item = Result<String, std::io::Error>>>(
+    iter: &mut Peekable<T>,
+    name: &str,
+) -> ProcResult<Option<usize>> {
+    if iter
+        .peek()
+        .map(|line| line.as_ref().is_ok_and(|line| line.contains("geniv")))
+        .unwrap_or(false)
+    {
+        let val = parse_line(iter, "geniv", name)?;
+        if val != "<none>" {
+            return Ok(Some(expect!(usize::from_str(&val))));
+        }
+    }
+    Ok(None)
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn parse_line_correct() {
+        let line = Ok("name         : ghash".to_string());
+        let mut iter = std::iter::once(line);
+        let val = match parse_line(&mut iter, "name", "parse_line_correct") {
+            Ok(val) => val,
+            Err(e) => panic!("{}", e),
+        };
+        assert_eq!("ghash", val);
+    }
+
+    #[test]
+    fn parse_line_incorrect() {
+        let line = Ok("name         : ghash".to_string());
+        let mut iter = std::iter::once(line);
+        let val = match parse_line(&mut iter, "name", "parse_line_incorrect") {
+            Ok(val) => val,
+            Err(e) => panic!("{}", e),
+        };
+        assert_ne!("hash", val);
+    }
+
+    #[test]
+    fn parse_block() {
+        let block = r#"driver       : deflate-generic
+module       : kernel
+priority     : 0
+refcnt       : 2
+selftest     : passed
+internal     : no
+type         : compression"#;
+        let mut iter = block.lines().map(|s| Ok(s.to_string())).peekable();
+        let block = CryptoBlock::from_iter(&mut iter, "deflate");
+        assert!(block.is_ok());
+        let block = block.unwrap();
```

Comment:
The assertion on `block.is_ok()` is redundant, because `block` gets unwrapped on the next line, so you can just remove this line.

```suggestion
        let block = block.unwrap();
```

Reply:
Hey, I've modified to be an expect, just so we can have an error message in there to make it easier to understand the context when the test fails.